### PR TITLE
chore: CI improvements

### PR DESCRIPTION
**Pull request by tadejsv**
- [x] Move black, flake8, isort configuration to `pyproject.toml` and `setup.cfg`
- [x] Update CI and pre-commit hooks

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -4,86 +4,48 @@ on:
...
-          CHANGED_FILES: ${{ steps.file_changes.outputs.added_modified }}
+      - name: Install requirements
+        run: pip install -r .github/requirements-test.txt
+      - name: check isort
+        run: isort --check .
 
   prep-testbed:
     runs-on: ubuntu-latest
-    needs: [commit-lint, lint-flake-8, check-black]
+    needs: [lint-flake-8, check-black, check-isort]
```
Comment (bwanglzu) [inline .github/workflows/ci.yml:48]: i think a general better thing to do is to only apply changes on "updated files within the PR", not all files.

**Review by bwanglzu — COMMENTED** [review]
quick comment, will continue a bit later

File: .github/workflows/ci.yml (hunk 1 above)
Comment (tadejsv) [reply .github/workflows/ci.yml:48]: If you start from a state where all checks pass, it is the same. Running time is negligible (<1s for all), so I don't see a good reason why we need it - it just introduces an extra step in CI

**Review by bwanglzu — APPROVED** [review]
LGTM!

**Pull request merged**